theano/compile/tests/test_builders.py

```diff
@@ -151,6 +152,14 @@ def test_connection_pattern(self):
                          [True, False, True]]
         assert results == expect_result
 
+    def test_infer_shape(self):
+        x = T.matrix('x')
+        y = x+x
+        op_graph = OpFromGraph([x], [y], mode='FAST_RUN')
```

Comment:
try without it and run the test, by replacing theano-nose or nosetests by:

theano/misc/do_nightly_buildbot

this will test with many modes. Do this before updating the PR.